Week4/ReadingList/app/src/main/java/com/florencenjeri/readinglist/LogInFragment.kt

```diff
@@ -0,0 +1,70 @@
+package com.florencenjeri.readinglist
+
+import android.os.Bundle
+import android.text.TextUtils
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import androidx.navigation.fragment.findNavController
+import com.florencenjeri.readinglist.model.UserPrefs
+import kotlinx.android.synthetic.main.fragment_log_in.*
+
+class LogInFragment : Fragment() {
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        setHasOptionsMenu(false)
+        //Check if the user is logged in
+        if (UserPrefs.isUserLoggedIn()) {
+            findNavController().navigate(LogInFragmentDirections.actionLogInFragmentToBooksFragment())
+        }
+        // Inflate the layout for this fragment
+        return inflater.inflate(R.layout.fragment_log_in, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        buttonLogIn.setOnClickListener {
+            if (isValidated()) {
+                UserPrefs.logInUser(editTextEmail.text.toString(), editTextPassword.text.toString())
+                findNavController().navigate(LogInFragmentDirections.actionLogInFragmentToBooksFragment())
+            }
+
+        }
+    }
+
+    fun isValidated(): Boolean {
+
+        var isValid = true
```

Comment:
I'd rename this to `validateData()` and remove the return type.

Or leave the return type, and do the rest of the code depending on which piece of information is missing, in different functions.

E.g. create a sealed class type named: `ValidationResult`, and then have cases such as `Valid, UsernameInvalid, PasswordInvalid` etc.